internal/engine/wazevo/backend/isa/arm64/instr_encoding.go

```diff
@@ -855,6 +869,48 @@ func encodeAluRRImm(op aluOp, rd, rn, amount, _64bit uint32) uint32 {
 	return _64bit<<31 | opc<<29 | 0b100110<<23 | _64bit<<22 | immr<<16 | imms<<10 | rn<<5 | rd
 }
 
+func encodeVecRRR(op vecOp, rd, rn uint32, arr vecArrangement) uint32 {
+	switch op {
+	case vecOpUaddlv:
+		q, size := vecArrangementParams(arr)
+		return q<<29 | 0b101110<<24 | size<<22 | 0b110000001110<<10 | rn<<5 | rd
+	default:
+		panic("unsupported or illegal vecOp: " + op.String())
+	}
+}
+
+func encodeVecMisc(op vecOp, rd, rn uint32, arr vecArrangement) uint32 {
```

Comment:
I can't believe it but I think I got both the encodings right on my first shot 🎉 
need to add unit tests though